dashboard/src/services/billing.ts

```diff
@@ -0,0 +1,65 @@
+import { getUserSubscription } from '@/repositories/postgres/subscription';
+import { getUserSiteIds } from '@/repositories/postgres/dashboard';
+import { getUserEventCountForPeriod } from '@/repositories/clickhouse/usage';
+import { getBillingHistoryByUserId } from '@/repositories/postgres/billingHistory';
+import { toDateString } from '@/utils/dateFormatters';
+import {
+  UserBillingDataSchema,
+  type UsageData,
+  type UserBillingData,
+  type BillingHistory,
+} from '@/entities/billing';
+
+export async function getUserBillingStats(userId: string): Promise<UserBillingData> {
+  try {
+    const subscription = await getUserSubscription(userId);
+    if (!subscription) {
+      throw new Error('No subscription found for user');
+    }
+
+    const siteIds = await getUserSiteIds(userId);
+
+    const currentUsage = await getUserEventCountForPeriod(siteIds, toDateString(subscription.currentPeriodStart));
+
+    const usage: UsageData = {
+      current: currentUsage,
+      limit: subscription.eventLimit,
+      remaining: Math.max(0, subscription.eventLimit - currentUsage),
+      isOverLimit: currentUsage >= subscription.eventLimit,
+      usagePercentage: (currentUsage / subscription.eventLimit) * 100,
+      daysUntilReset: getDaysUntilReset(subscription.currentPeriodEnd),
+      billingPeriod: {
+        start: subscription.currentPeriodStart,
+        end: subscription.currentPeriodEnd,
+      },
+    };
+
+    const isExistingPaidSubscriber = subscription.pricePerMonth > 0 && subscription.status === 'active';
```

Comment:
Question: I've seen somewhere that if `price < 0`, then it's a custom plan - this check will probably make custom plan users defined as not a paying subscriber? 🤔